jsonrpc/server.go

```diff
@@ -438,7 +431,10 @@ func (s *Server) handleRequest(ctx context.Context, req *Request) (*response, er
 	if errAny := tuple[1].Interface(); !isNil(errAny) {
 		res.Error = errAny.(*Error)
 		if res.Error.Code == InternalError {
-			s.listener.OnRequestFailed(req.Method, err)
+			s.listener.OnRequestFailed(req.Method, res.Error)
+			reqJSON, _ := json.Marshal(req)
+			errJSON, _ := json.Marshal(res.Error)
+			s.log.Debugw("failed handing RPC request", "req", string(reqJSON), "res", string(errJSON))
```

Comment:
If `reqJSON` or `errJSON` are `nil`, we can't convert them using `string(nil)`. I think we should set them to `[]byte{}` if `err != nil`.

Reply:
Actually, `string([]byte(nil))` just returns an empty string. So we should be good.